spec/globus/client_spec.rb

```diff
@@ -88,45 +83,68 @@
     end
   end
 
-  describe ".mkdir" do
-    it "invokes mkdir on the Endpoint class, injecting config" do
-      described_class.mkdir(user_id:, work_id:, work_version:)
-      expect(fake_endpoint).to have_received(:mkdir).once
+  describe "public class methods" do
+    let(:fake_instance) do
+      instance_double(
+        described_class,
+        mkdir: nil,
+        file_count: nil,
+        total_size: nil
+      )
     end
```

Comment:
When testing the class methods, we don't even need to care about the endpoint; we only need to care that these methods called on the class are invoked on the instance.